### PR TITLE
Add support for network gateways

### PR DESCRIPTION
**Pull request by sethvargo**
Spawned out of https://github.com/mitchellh/vagrant/pull/2870 with support for more OSes and configurations.

### CODE REVIEW

**Comment by sethvargo** [comment]
@mitchellh @phinze can I get a quick review here?

File: website/docs/source/v2/networking/private_network.html.md [hunk 1]
```diff
@@ -63,6 +63,9 @@ the [reserved private address space](http://en.wikipedia.org/wiki/Private_networ
 and most routers actually block traffic from going to them from the
 outside world.
 
+For some operating systems, additional configuration options for the static
+IP address are available such as setting the default gateway for mtu.
```
Comment (phinze) [inline website/docs/source/v2/networking/private_network.html.md]: `s/for mtu/or MTU/`, I think?

File: test/unit/templates/guests/arch/network_dhcp_test.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+require "vagrant/util/template_renderer"
+
+describe "templates/guests/arch/network_dhcp" do
+  let(:template) { "guests/arch/network_dhcp" }
+
+  it "renders the template" do
+    result = Vagrant::Util::TemplateRenderer.render(template, options: {
+      device: "en0",
+    })
+    expect(result).to eq <<-EOH.gsub(/^ {6}/, "")
```
Comment (phinze) [inline test/unit/templates/guests/arch/network_dhcp_test.rb]: <3 I love the indented heredoc style. We're going to get along well. :grin:

**Comment by phinze** [comment]
1 typo - otherwise LGTM!

**Comment by sethvargo** [comment]
Thanks!

**Pull request merged**